### PR TITLE
Article: Java Arrays

### PR DESCRIPTION
**Pull request by nag786**
This is quick reference to array data structure in java

### CODE REVIEW

**Comment by Rafase282** [comment]
Thanks for the article, just so you know, you should be able to add `java` to the code block for syntax highlight.

Also I would change the filename to `Java-Array.md`

File: Java-array.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Java Array
+Array is used to store a collection of data of similar datatype. Array always starts with 0 index.
+
+## Syntax:
+    dataType[] name_of_array;   // preferred way.
```
Comment (alayek) [inline Java-array.md]: Use Java specific Markdown code highlight syntax

File: Java-array.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    or
+    dataType name_of_array[];  //  works but not preferred way
+
```
Comment (alayek) [inline Java-array.md]: Can you add a line about why it's not preferred way?

File: Java-array.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+##### Another way to create an Array:
+
+    dataType[] name_of_array = {value0, value1, ..., valuek};
+    
+
+##### Code snippets of above syntax:
+    double[] list = {1, 2, 3, 4};
+
+- Example of code:
+```
```
Comment (alayek) [inline Java-array.md]: Use Java specific syntax highlighting.

```diff
- ```
+ ```java
```

File: Java-array.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ int[] a = new int[] {4,5,6,7,8};//declare array
+for (int i=0; i<a.length; i++)//loop goes through each index
+{
+  System.out.println(a[i]);// prints the array
+
+}
+```
```
Comment (alayek) [inline Java-array.md]: Add a [REPL](https://repl.it) snippet to it.

File: Java-array.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Output:
+```
+4
+5
+6
+7
+8
+```
+
+Source: [Java Array](http://www.tutorialspoint.com/java/java_arrays.htm)
```
Comment (alayek) [inline Java-array.md]: Refer to official Oracle documentation.

**Comment by alayek** [comment]
Thanks @nag786 for this article! I have left some comments, please check.

@koustuvsinha please take a look!

File: Java-array.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+# Java Array
+Array is used to store a collection of data of similar datatype. Array always starts with 0 index.
```
Comment (theflametrooper) [inline Java-array.md]: ~~Array~~ An Array

File: Java-array.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+## Syntax:
+    dataType[] name_of_array;   // preferred way.
+    or
+    dataType name_of_array[];  //  works but not preferred way
+
+##### code snippets of above syntax:
```
Comment (theflametrooper) [inline Java-array.md]: ~~code snippets of above syntax:~~ code snippets of the above syntax:

**Comment by koustuvsinha** [comment]
Thanks @nag786 ! I appreciate your effort but Java Arrays is a big area, and you have missed out on a lot of edge cases / methods in your article. It's ok, we are also no experts here. Look [at this article](http://tutorials.jenkov.com/java/arrays.html) for reference. See how the author (the best Java article writer out there) structures the article, includes the edge cases and tit-bits. Now you don't have to copy the article, we also don't promote that, but try to follow a similar pattern. Write in your own words, provide appropriate code snippets along with [REPL](https://repl.it/) snippets (as much code possible the better). See our other Java articles.

Writing tech articles is not a simple task, it requires patience, dedication, and the passion to teach someone something new! Imagine you are teaching your friend who is completely new in Java about Arrays, and write like that. Sorry for my long pedantic talk, but since you are new _and_ we want high quality articles so trying my best to get you onboard! Talk with us on Gitter if you have any queries any time!

Update this PR only, and let us know. :)

**Comment by theflametrooper** [comment]
@nag786 When do you plan to update this article with the suggested changes?

**Comment by nag786** [comment]
@koustuvsinha  Thank you for your time.You are a perfectionist. When I get to his age, I would be writing   like him.I will try to do my best. I 'll add more items to Arrays.
@alayek Thanks for correcting errors.
@Rafase282 Thanks
@theflametrooper  Thanks for  correcting grammar.I will update soon but I need more time.
Why people are changing  my given file name?
If you start with the word Java first or last , what is the difference?

**Comment by theflametrooper** [comment]
@nag768 We keep all related articles in a similar file name structure for continuity. Good luck with the changes!

**Comment by nag786** [comment]
Thanks. That makes sense.

**Comment by alayek** [comment]
@nag786 about your question on file names, let me add on top of what @theflametrooper just said. Our Wiki articles are not always accessed by users by directly coming to a GitHub Wiki.

Some of it is fetched by @camperbot on demand in chat rooms. Some of it is also fetched within the main site by using the "Wiki" link on navbar. In future, there could be more ways to access them.

Keeping the same standard helps us write RegEx more easily, when our programs are accessing these. Now if you suddenly break from the pattern, we don't know what could break.

It's better to just follow the naming convention.

**Comment by nag786** [comment]
@alayek  Thanks for the explanation. Now I got it.

**Comment by theflametrooper** [comment]
@nag786 Will you have this finished in the next 2 days? We're hoping to have this PR merged by Friday.

**Pull request merged**